rewrite-gradle/src/main/java/org/openrewrite/gradle/ChangeDependencyArtifactId.java

```diff
@@ -112,7 +113,10 @@ public G visitCompilationUnit(G.CompilationUnit compilationUnit, ExecutionContex
             @Override
             public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
                 J.MethodInvocation m = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
-                if (!dependencyDsl.matches(m) || !(StringUtils.isBlank(configuration) || m.getSimpleName().equals(configuration))) {
+
+                GradleDependency.Matcher gradleDependencyMatcher = new GradleDependency.Matcher();
+                
+                if (!gradleDependencyMatcher.get(getCursor()).isPresent() || !(StringUtils.isBlank(configuration) || m.getSimpleName().equals(configuration))) {
```

Comment:
An additional enhancement here would be to expose `configuration` as another selector for the matcher. This isn't strictly necessary as part of this PR though, but I wanted to mention it while I was here.